sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -81,52 +81,42 @@ private string GetSessionFilePath()
 
             string fileName = name + (IsAsync ? "Async" : string.Empty) + ".json";
 
-            var path = TestContext.CurrentContext.TestDirectory;
-            var indexartifacts = path.IndexOf("artifacts");
-            if (indexartifacts == -1)
-            {
-                return Path.Combine(TestContext.CurrentContext.TestDirectory,
-                "SessionRecords",
-                additionalParameterName == null ? className : $"{className}({additionalParameterName})",
-                fileName);
-            }
-            var newpath = path.Substring(0, indexartifacts) + "sdk";
+            System.Collections.Generic.IEnumerable<System.Attribute> customAttributes = this.GetType().Assembly.GetCustomAttributes(typeof(AssemblyMetadataAttribute));
 
-            // check if Track 1 or Track 2
-            int firstIndex = path.IndexOf("bin\\");
-            int lastIndex = path.LastIndexOf(".Tests");
-            if (firstIndex == -1 || lastIndex == -1)
+            Dictionary<string, string> attributes = new Dictionary<string, string>();
+            foreach (var attribute in customAttributes)
             {
-                return Path.Combine(TestContext.CurrentContext.TestDirectory,
-                "SessionRecords",
-                additionalParameterName == null ? className : $"{className}({additionalParameterName})",
-                fileName);
+                attributes.Add(((AssemblyMetadataAttribute) attribute).Key, ((AssemblyMetadataAttribute) attribute).Value);
```

Comment:
Why do we store all the key value pairs, rather than just grabbing the ones we need?